src/tools/rbd_mirror/image_map/LoadRequest.cc

```diff
@@ -80,7 +82,87 @@ void LoadRequest<I>::handle_image_map_list(int r) {
     return;
   }
 
-  finish(0);
+  mirror_image_list();
+}
+
+template<typename I>
+void LoadRequest<I>::mirror_image_list() {
+  dout(20) << dendl;
+
+  librados::ObjectReadOperation op;
+  librbd::cls_client::mirror_image_list_start(&op, m_start_after, MAX_RETURN);
+
+  m_out_bl.clear();
+  librados::AioCompletion *aio_comp = create_rados_callback<
+    LoadRequest<I>,
+    &LoadRequest<I>::handle_mirror_image_list>(this);
+  int r = m_ioctx.aio_operate(RBD_MIRRORING, aio_comp, &op, &m_out_bl);
+  ceph_assert(r == 0);
+  aio_comp->release();
+}
+
+template<typename I>
+void LoadRequest<I>::handle_mirror_image_list(int r) {
+  dout(20) << ": r=" << r << dendl;
+
+  if (r < 0) {
```

Comment:
nit: this check for an error looks like not necessary here, because we still check this later after decoding.

Reply:
Ah oups yes indeed